src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +40,37 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${checkedTodo?.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => {
+                closeModal(null);
+                resetTodoClickedState();
+              }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)